p2p/dial.go

```diff
@@ -460,10 +454,10 @@ func (d *dialScheduler) startDial(task *dialTask) {
 	hkey := string(task.dest.ID().Bytes())
 	d.history.add(hkey, d.clock.Now().Add(dialHistoryExpiration))
 	d.dialing[task.dest.ID()] = task
-	gopool.Submit(func() {
+	go func() {
 		task.run(d)
```

Comment:
Use time out context here to limit un-endless dial?

Reply:
by default TCP dial exist [OS level timeout](https://pkg.go.dev/net#Dialer), maybe wait upstream merge ?
https://github.com/dogechain-lab/dbsc/blob/691902b3b07786521dbc0766c357277a3c0eb370/p2p/dial.go#L66-L68

Reply:
Not necessary.
We'll leave it here if it does expire.